new/rfcXXX-bootstrap-partial.md

```diff
@@ -0,0 +1,49 @@
+---
+RFC: unassigned
+Author: Ranjib Dey (ranjib@linux.com)
+Status: Draft
+Type: Standards Track
+---
+
+# Allow partials in bootstrap template
+
+Allow partial templates to be passed during `knife bootstrap`
+
+## Motivation
+
+Chef allows knife based bootstrapping via standard templates as well
+as custom templates. But theres no way to reuse the bootstrapping logic
+of standard templates. For minimal distributions where tools like
+wget and curl is absent, one has to roll his/her own bootstrap template,
+which most often is a copy of the standard template with extra commands
+to install those tools. Over the time, this becomes tech debt, as chef evolves
+and with it the standard template changes, incorporating new bootstrap options.
+
+Scope of this RFC is to offer users a mechnism to execute custom logic before
+the standard bootstrapping process, instead of specifying the entire bootstrapping
```

Comment:
This sounds like a separate template that runs before the bootstrap template. An example was given that you could ```apt-get update && apt-get install wget```. In that case, I think this should be described as a pre-template.

If we want true partials, I'd want a way we can describe the chef-full template into steps, and it would be best if we could agree on those steps in the RFC. For instance, right now the templates call the "start_chef" step which is defined in core/bootstrap_context.rb instead of the template. The partial should be able to replace a step. There could be a pre step that does nothing in the base template that a partial could hook. There could be a start_chef step. Maybe a config_files step and a install step.

For discussion:

* preinstall (empty)
* install
* postinstall (empty)
* config_files
* start
* finish (empty)